File: Linked Lists/mergeTwoLinkedLists.py
# Singly-linked lists are already defined with this interface:
# class ListNode(object):
#   def __init__(self, x):
#     self.value = x
#     self.next = None
#
def solution(l1, l2):
    result = ListNode(0)
    _result = result
    while(l1 or l2):
        if l1 == None:
            _result.next = ListNode(l2.value)
            l2 = l2.next
        elif l2 == None:
            _result.next = ListNode(l1.value)
            l1 = l1.next
        else:
            if l1.value >= l2.value:
                _result.next = ListNode(l2.value)
                l2 = l2.next
            else:
                _result.next = ListNode(l1.value)
                l1 = l1.next
        _result = _result.next
    return result.next
        

File: Linked Lists/mergeTwoLinkedLists.php
// Singly-linked lists are already defined with this interface:
// class ListNode {
//   public $value;
//   public $next;
//
//   public function __construct($x) {
//     $this->value = $x;
//     $this->next = null;
//   }
// }
//
function solution($l1, $l2) {
    $result = new ListNode(0);
    $_result = $result;
    while ($l1 || $l2){
        if(!$l1){
            $_result->next = $l2;
            $l2 = null;
        }else if(!$l2){
            $_result->next = $l1;
            $l1 = null;
        }else{
            if($l1->value >= $l2->value){
                $_result->next = new ListNode($l2->value);
                $l2 = $l2->next;
            }else{
                $_result->next = new ListNode($l1->value);
                $l1 = $l1->next;
            }
        }
        $_result = $_result->next;
    }
    return $result->next;
}
